pkg/autodiscover/autodiscover_podset.go

```diff
@@ -71,54 +82,71 @@ func findDeploymentByLabel(
 		if len(dps.Items) == 0 {
 			log.Warn("Did not find any deployments in ns=%s", ns)
 		}
-
 		for i := 0; i < len(dps.Items); i++ {
-			for _, aLabelObject := range labels {
-				log.Debug("Searching pods in deployment %q found in ns %q using label %s=%s", dps.Items[i].Name, ns, aLabelObject.LabelKey, aLabelObject.LabelValue)
-				if dps.Items[i].Spec.Template.ObjectMeta.Labels[aLabelObject.LabelKey] == aLabelObject.LabelValue {
-					deployments = append(deployments, dps.Items[i])
-					log.Info("Deployment %s found in ns=%s", dps.Items[i].Name, ns)
+			if len(labels) > 0 {
+				if isDeploymentsPodsMatchingAtLeastOneLabel(labels, ns, &dps.Items[i]) {
+					allDeployments = append(allDeployments, dps.Items[i])
+					continue
 				}
+			} else {
+				// If labels are not provided in the namespace under test, they are tested by the CNF suite
```

Comment:
"If labels are not provided in the namespace under test, they are tested by the CNF suite" -> "If labels are not provided, all deployments in the namespaces under test, are tested by the CNF suite"